solr/core/src/java/org/apache/solr/search/combine/ReciprocalRankFusion.java

```diff
@@ -69,21 +68,24 @@ public int getK() {
   @Override
   public QueryResult combine(List<QueryResult> rankedLists, SolrParams solrParams) {
     int kVal = solrParams.getInt(CombinerParams.COMBINER_RRF_K, this.k);
-    List<DocListAndSet> docListAndSet = getDocListsAndSetFromQueryResults(rankedLists);
     QueryResult combinedResult = new QueryResult();
-    combineResults(combinedResult, docListAndSet, false, kVal);
+    combineResults(combinedResult, rankedLists, false, kVal);
     return combinedResult;
   }
 
-  private static List<DocListAndSet> getDocListsAndSetFromQueryResults(
-      List<QueryResult> rankedLists) {
-    List<DocListAndSet> docLists = new ArrayList<>(rankedLists.size());
-    for (QueryResult rankedList : rankedLists) {
-      docLists.add(rankedList.getDocListAndSet());
-    }
-    return docLists;
-  }
-
+  /**
+   * Merges per-shard ranked results using Reciprocal Rank Fusion (RRF).
```

Comment:
Reading this sentence, I think okay this is a typical application of RRF.  And the code does that, so it's technically not wrong.  But lets be real with the reader; in practice this is going to interleave the results because docs uniquely only appear in one shard.  Say that.  And say that you change the score to the RRF score.

As an aside, it'd probably be ideal to secondarily sort based on doc ID so we have more reproducibility.  WDYT?